novice-r-practice.Rmd

```diff
@@ -0,0 +1,375 @@
+
+6. Read `visited.csv`, drop the NAs and store in `visits`.  Use `anti_join()` to find the measurements in `cleaned` that don't have matches in `visits`:
+
+```{r anti-join-two-tables}
+visits <- readr::read_csv(here::here("novice-r-practice", "visited.csv")) %>%
+  dplyr::filter(!is.na(visit_date)) 
+cleaned %>% anti_join(visits)
+```
+
+*Are there any sites in `visits` that don't have matches in `cleaned`?*
+
+7. Join `visits` with the cleaned-up table of readings:
+
+```{r join-two-tables}
+cleaned <- visits %>% 
+  dplyr::inner_join(cleaned, by = c("visit_id" = "visit_id"))
+cleaned
+```
+
+*Join `visited.csv` with `site.csv` to get (date, latitude, longitude) triples for site visits.*
+
+7. Find the dates of the highest radiation reading at each site:
+
+```{r dates-high-rad}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(max_rad = max(reading)) %>%
+  dplyr::filter(reading == max_rad)
+```
+
+Another way to do it:
+
+```{r dates-high-rad-2}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::top_n(1, reading) %>%
+  dplyr::select(site_id, visit_date, reading)
+```
+
+*Explain why this __doesn't__ work.*
+
+```{r error-high-rad, error=TRUE}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::summarize(max_rad = max(reading)) %>%
+  dplyr::ungroup() %>%
+  dplyr::filter(reading == max_rad)
+```
+
+8. Normalize radiation against the highest radiation seen per site:
+
+```{r normalize-rad}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(
+    max_rad = max(reading),
+    frac_rad = reading / max_rad) %>%
+  dplyr::select(visit_id, site_id, visit_date, frac_rad)
+```
+
+*Normalize salinity against mean salinity by site.*
+
+9. Find stepwise change in radiation per site by date:
+
+```{r rad-change}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(delta_rad = reading - dplyr::lag(reading)) %>%
+  dplyr::arrange(site_id, visit_date)
+```
+
+*Find length of time between visits by site.*
+
+10. Find sites that experience any stepwise increase in radiation between visits:
+
+```{r rad-increases}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(delta_rad = reading - dplyr::lag(reading)) %>%
+  dplyr::filter(!is.na(delta_rad)) %>%
+  dplyr::summarize(any_increase = any(delta_rad > 0)) %>%
+  dplyr::filter(any_increase)
+```
+
+*Find sites with visits more than one year apart.*
+
+## Creating charts
+
+We will use data on the mass and home range area (HRA) of various species from:
+
+> Tamburello N, Côté IM, Dulvy NK (2015) Data from: Energy and the scaling of animal space use. Dryad Digital Repository.
+> https://doi.org/10.5061/dryad.q5j65
+
+```{r}
+hra <- readr::read_csv(here::here("novice-r-practice", "home-range-database.csv"))
+head(hra)
```

Comment:
Given that `read_csv()` reads in the data as a `tibble()`, you don't need to use `head()` tibbles automatically only output the first 6 or so rows.